x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -69,6 +67,80 @@ export function insertNewColumn({
     }
   }
 
+  if (operationDefinition.input === 'fullReference') {
+    let tempLayer = { ...layer };
+    const referenceIds = operationDefinition.requiredReferences.map((validation) => {
+      const validOperations = Object.values(operationDefinitionMap).filter(({ type }) =>
+        isOperationAllowedAsReference({ validation, operationType: type })
```

Comment:
I guess we should pass in the field we get via parameter as well here? Otherwise `isOperationAllowedAsReference` will never be called with a field

Reply:
I'll clarify this in a comment in code, but basically I'm not expecting the `field` to be passed in while constructing a reference-based operation. However I would expect that the `operationSupportMatrix` would be used here to look for cases where there's only one possible field or operation.